### PR TITLE
Improve TeachingBubble usage docs

### LINKED ISSUE #7243: TeachingBubble: Usage documentation is woefully inadequate
### Environment Information 
- __Package version(s)__: @uifabric/fabric-website@6.8.0
- __Browser and OS versions__: N/A

### Please provide a reproduction of the bug in a codepen:
 The docs can be seen on both the live Fabric site: https://developer.microsoft.com/en-us/fabric#/components/teachingbubble

And on the dev site (here at the 6.106.3 snap): https://fabricweb.z5.web.core.windows.net/oufr/6.106.3/#/examples/teachingbubble
 
#### Actual behavior:
The usage docs for TeachingBubble are essentially absent, and don't help consumers differentiate it from similar components like Callout, Tooltip, or Coachmark. The TeachingBubble is a very specific kind of Callout that is meant to be used in tandem with a Coachmark, to call attention to a part of a page. It was specifically designed to educate users about new features, and shouldn't be thought of as a helper for assistive text.

This can lead to cases where the control is used in contexts that it wasn't designed for, and where another control may be more appropriate.
 
#### Expected behavior:
The usage docs for TeachingBubble should be expanded, and should likely borrow from Coachmark to differentiate its use case.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Normal
 
Products/sites affected: N/A

### PR DESCRIPTION
**Pull request by Jahnp**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #7243
- [x] Include a change request file using `$ npm run change`

#### Description of changes
See #7243. TeachingBubble's usage docs are woefully inadequate; this PR expands on them. It also fixes a small visual bug on one of the examples.

### CODE REVIEW

**Comment by Jahnp** [comment]
@bigbadcapers , would you mind taking a look as well when you have a chance? Thank you!

File: packages/office-ui-fabric-react/src/components/TeachingBubble/examples/TeachingBubble.WideIllustration.Example.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class TeachingBubbleWideIllustrationExample extends React.Component<{}, I
 
   public render(): JSX.Element {
     const { isTeachingBubbleVisible } = this.state;
-    const exampleImageProps: IImageProps = { src: 'http://placehold.it/364x140' };
+    const exampleImageProps: IImageProps = { src: 'http://placehold.it/154x140' };
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/TeachingBubble/examples/TeachingBubble.WideIllustration.Example.tsx]: You rock!

File: packages/office-ui-fabric-react/src/components/TeachingBubble/docs/TeachingBubbleDonts.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- Don't use TeachingBubbles for user-invoked actions.
+- Don't use TeachingBubbles when a Callout would be sufficient.
+- Don't show a TeachingBubble more than once, even if the user has not completed the action.
```
[the remark this answers is not in the document]
Comment (Jahnp) [reply packages/office-ui-fabric-react/src/components/TeachingBubble/docs/TeachingBubbleDonts.md]: That's a good callout (har har). Ben makes a good point elsewhere about only showing TeachingBubbles temporarily, not permanently--I think that relates here (e.g. should only be shown while a new feature is still new). I'll revisit and update this. Thanks for the feedback!

[the remark this answers is not in the document]
Comment (Jahnp) [reply packages/office-ui-fabric-react/src/components/TeachingBubble/docs/TeachingBubbleDonts.md]: Great points! We also chatted in person about bringing in more of the positive Dos from the overview. I'll incorporate those.

File: packages/office-ui-fabric-react/src/components/TeachingBubble/docs/TeachingBubbleDonts.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+- Don't use TeachingBubbles for user-invoked actions.
```
[the remark this answers is not in the document]
Comment (Jahnp) [reply packages/office-ui-fabric-react/src/components/TeachingBubble/docs/TeachingBubbleDonts.md]: Your 2nd paragraph is 100% right. They should generally be shown on initial render to draw attention to something important, but they're not meant as a supplemental component like Callouts. The lines could get a little blurry, but that's ok :) 

Great point about adding an example. I'll clarify.

**Comment by bigbadcapers** [comment]
@Jahnp thanks for adding me, in terms of documentation the new details looks good. With regards to the animation itself, the ask is not to make it disable-able but rather improve it because it's got weird framerate issues and slowdown issues towards the end. The original design intent is to match this bouncejs: https://goo.gl/4pm5mA

**Pull request merged**